apps/tangle-dapp/src/components/account/RewardsAndPoints.tsx

```diff
@@ -0,0 +1,117 @@
+import { useActiveChain } from '@webb-tools/api-provider-environment/hooks/useActiveChain';
+import { ZERO_BIG_INT } from '@webb-tools/dapp-config/constants';
+import { RestakeAssetId } from '@webb-tools/tangle-shared-ui/types';
+import { EMPTY_VALUE_PLACEHOLDER } from '@webb-tools/webb-ui-components/constants';
+import { Typography } from '@webb-tools/webb-ui-components/typography/Typography';
+import get from 'lodash/get';
+import { useMemo } from 'react';
+import { formatUnits, zeroAddress } from 'viem';
+import useAssetsMetadata from '../../data/assets/useAssetsMetadata';
+import useAccountRewardInfo from '../../data/rewards/useAccountRewardInfo';
+import KeyStatsItem from '../KeyStatsItem/KeyStatsItem';
+import ClaimRewardAction from './ClaimRewardAction';
+
+const RewardsAndPoints = () => {
+  const [activeChain] = useActiveChain();
+
+  const { result, isLoading, error } = useAccountRewardInfo();
+
+  const {
+    result: assetsMetadata,
+    isLoading: isAssetsMetadataLoading,
+    error: assetsMetadataError,
+  } = useAssetsMetadata(
+    useMemo(() => Object.keys(result ?? {}) as RestakeAssetId[], [result]),
+  );
+
+  const displayRewardAsset = useMemo(() => {
+    if (result === null || !activeChain) {
+      return null;
+    }
+
+    const substrateNativeBalance = get(result, `${ZERO_BIG_INT}`, null);
+    const hasSubstrateNativeBalance =
+      substrateNativeBalance !== null &&
+      substrateNativeBalance !== ZERO_BIG_INT;
+
+    const evmNativeBalance = get(result, zeroAddress, null);
+    const hasEvmNativeBalance =
+      evmNativeBalance !== null && evmNativeBalance !== ZERO_BIG_INT;
+
+    // Return the first non-null balance
+    if (!hasEvmNativeBalance && !hasSubstrateNativeBalance) {
+      const representativeAsset = Object.entries(result).find(
+        ([assetId, balance]) => {
+          const hasBalance = balance > ZERO_BIG_INT;
+          const hasMetadata = assetsMetadata.get(assetId as RestakeAssetId);
```

Comment:
Avoid casting and instead use runtime assertions to catch future logic bugs which may otherwise go silent and become harder to debug:
```suggestion
          const hasMetadata = assetsMetadata.get(assertRestakeAssetId(assetId));
```